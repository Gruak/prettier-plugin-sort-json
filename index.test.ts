import { format } from 'prettier';
import * as SortJsonPlugin from '.';

describe('Sort JSON', () => {
  it('should validate sorted JSON', () => {
    const fixture = {
      0: '0',
      a: 'a',
      b: 'b',
      c: 'c',
    };

    const input = JSON.stringify(fixture, null, 2);
    const output = format(input, {
      filepath: 'foo.json',
      parser: 'json',
      plugins: [SortJsonPlugin],
    });

    expect(output).toMatchSnapshot();
  });

  it('should sort unsorted JSON', () => {
    const fixture = {
      0: '0',
      b: 'b',
      a: 'a',
      c: 'c',
    };

    const input = JSON.stringify(fixture, null, 2);
    const output = format(input, {
      filepath: 'foo.json',
      parser: 'json',
      plugins: [SortJsonPlugin],
    });

    expect(output).toMatchSnapshot();
  });
});
