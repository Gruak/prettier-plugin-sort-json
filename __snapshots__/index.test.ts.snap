// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort JSON should sort unsorted JSON 1`] = `
"{
  \\"0\\": \\"0\\",
  \\"a\\": \\"a\\",
  \\"b\\": \\"b\\",
  \\"c\\": \\"c\\"
}
"
`;

exports[`Sort JSON should validate sorted JSON 1`] = `
"{
  \\"0\\": \\"0\\",
  \\"a\\": \\"a\\",
  \\"b\\": \\"b\\",
  \\"c\\": \\"c\\"
}
"
`;
